### PR TITLE
Prompt user to either abort pr flow or ignore untracked changes

### PR DESCRIPTION
**Pull request by henrik-fauskanger-statnett**
Testing:
- [x] Unit Tests
- [ ] Integration Tests
- Test Command: make test

Documentation:
- No updates

### CODE REVIEW

**Comment by henrik-fauskanger-statnett** [comment]
This PR attempts to handle one of the possible edge cases when running `gh dxp pr`, namely that the local repo contains untracked files. Arcanist behaviour is that the user must either explicitly agree to ignoring these files or abort the pr creation, so this is what we're trying to replicate.

The case of having uncommitted, but tracked, files is slightly more complex and is not handled by this diff

File: pkg/pr/pr.go [hunk 1]
```diff
@@ -33,7 +34,53 @@ func Execute(exe utils.Executor, options *Options) error {
 	return create(exe, options, branchID)
 }
 
+func filter(list []string, test func(string) bool) (ret []string) {
```
Comment (henrik-fauskanger-statnett) [inline pkg/pr/pr.go:37]: This function is a very general utility function and should perhaps be moved somewhere else in the repo

**Review by johannes-gs — APPROVED** [review]
